mlir/include/mlir/Dialect/GPU/IR/CompilationAttrs.td

```diff
@@ -16,6 +16,155 @@
 include "mlir/Dialect/GPU/IR/GPUBase.td"
 include "mlir/Dialect/GPU/IR/CompilationAttrInterfaces.td"
 
+//===----------------------------------------------------------------------===//
+// GPU kernel attribute
+//===----------------------------------------------------------------------===//
+
+def GPU_KernelAttr : GPU_Attr<"Kernel", "kernel"> {
+  let description = [{
+    GPU attribute for storing metadata related to a compiled kernel. The
```

Comment:
The name of the attribute is "kernel" but it does not store a kernel, only metadata: is this the best name?

Reply:
What about `KernelMetadata`?

Reply:
LG